src/Viewer/Viewer.js

```diff
@@ -215,13 +227,28 @@ export function Viewer ({fileInfo, prettifyLog, logEventNumber, timestamp}) {
                 setStatusMessage("");
                 break;
             case CLP_WORKER_PROTOCOL.LOAD_LOGS:
+                if (resetFolding.current !== undefined) {
+                    resetFolding.current();
+                }
                 setLogData(event.data.logs);
                 setLoadingLogs(false);
                 setStatusMessage("");
                 break;
             case CLP_WORKER_PROTOCOL.UPDATE_FILE_INFO:
                 setFileMetadata(event.data.fileState);
                 break;
+            case CLP_WORKER_PROTOCOL.GET_SIMILAR_LINES:
+                if (addFolding.current !== undefined) {
+                    const range = event.data.state.range;
+                    const lines = range.end - range.start + 1;
+                    if (lines <= 1) {
+                        setStatusMessage("No similar logs found.");
+                    } else {
+                        setStatusMessage(`Collapsed ${lines} lines.`);
+                    }
+                    addFolding.current(range);
```

Comment:
I hope this can be the other way around. Instead of passing a callback function from `MonacoInstance.js` and calling it here in `Viewer.js` at the end of the workflow, I think it's more natural for `MonacoInstance.js` to post a query message to   `Viewer.js` in order to get a response that contains the range of similar lines, and then decide what to do from then on.